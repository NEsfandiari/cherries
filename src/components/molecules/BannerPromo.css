/** CSS from banner text - import Montserrat from Google Fonts
TODO - figure out how to use this font throughout the whole page */

@import url('https://fonts.googleapis.com/css?family=Montserrat');

.banner-text {
  font-family: 'Montserrat';
  font-style: italic;
  font-weight: bold;
}
