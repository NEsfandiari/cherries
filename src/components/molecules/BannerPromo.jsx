import React, { Component } from 'react'
import styled from 'styled-components'
import { IoIosClose } from 'react-icons/io'

const Container = styled.div`
  position: relative;
  top: -8px;
  width: 100%;
  display: flex;
  justify-content: start;
  align-items: left;
  background-color: #e20031;
  padding: 0.5rem;
  height: 2.5rem;
  text-align: left;
  p {
    margin-bottom: 0;
    margin-right: 1rem;
  }
  .banner-text {
    font-family: 'Montserrat';
    font-style: italic;
    font-weight: bold;
  }
`
const Announcement = styled.div`
  color: white;
  font-size: 20px;
  margin: 0px 15px 10px 20px;
  font-style: italic;
  font-weight: bold;
`
const PromoText = styled.div`
  color: white;
  font-size: 12px;
  margin-top: 2px;
  font-weight: 400;
  font-style: italic;
  font-weight: bold;
`
const Icon = styled.div`
  color: black;
  margin-left: auto;
  :hover {
    cursor: pointer;
  }
`
class BannerPromo extends Component {
  constructor(props) {
    super(props)
    this.state = {
      display: true,
    }

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick() {
    this.setState({ display: false })
  }

  render() {
    const banner = this.state.display ? (
      <Container>
        <Announcement className="banner-text">
          {this.props.bannerTitle}
        </Announcement>
        <PromoText className="banner-text">{this.props.bannerText}</PromoText>
        <Icon>
          <IoIosClose
            // style={{ color: 'black' }}
            size={30}
            onClick={this.handleClick}
          />
        </Icon>{' '}
      </Container>
    ) : (
      ''
    )
    return <React.Fragment>{banner}</React.Fragment>
  }
}

export default BannerPromo
