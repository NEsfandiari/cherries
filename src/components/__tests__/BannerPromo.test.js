import React from 'react'
import renderer from 'react-test-renderer'
import BannerPromo from '../molecules/BannerPromo'

describe('BannerPromo', () =>
  it('renders correctly', () => {
    const bannerTitle = 'Announcements!'
    const bannerText = 'NEW SALE!!'
    const tree = renderer
      .create(<BannerPromo bannerText={bannerText} bannerTitle={bannerTitle} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  }))
