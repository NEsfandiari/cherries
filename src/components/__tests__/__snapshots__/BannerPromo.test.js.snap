// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerPromo renders correctly 1`] = `
<div
  className="sc-bdVaJa cSiefN"
>
  <div
    className="banner-text sc-bwzfXH iAXSkC"
  >
    Announcements!
  </div>
  <div
    className="banner-text sc-htpNat ggROMQ"
  >
    NEW SALE!!
  </div>
  <div
    className="sc-bxivhb dXIVKd"
  >
    <svg
      fill="currentColor"
      height={30}
      onClick={[Function]}
      size={30}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width={30}
    >
      <path
        d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
      />
    </svg>
  </div>
   
</div>
`;
